### PR TITLE
Fix out-of-bounds access and remove MSVC function export

### PR DESCRIPTION
**Pull request by DomClark**
Partially reverts #4, and reverts #5. We decided to specify the function export using a linker flag, rather than an attribute in the code, so that the code continues to work with a vanilla `ladspa.h`. See https://github.com/LMMS/cmt/pull/5#issuecomment-2103323623 for details on the out-of-bounds fix.

### CODE REVIEW

**Review by tresf — APPROVED** [review]

**Review by Veratil — APPROVED** [review]
LGTM and tested MSVC build fixup with LMMS/lmms#6973 fix to CMakeLists.txt for cmt

File: src/delay.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ class DelayLine : public CMT_PluginInstance {
 
   DelayLine(const unsigned long lSampleRate,
 	    const LADSPA_Data fMaximumDelay) 
-    : CMT_PluginInstance(4),
+    : CMT_PluginInstance(5),
```
Comment (JohannesLorenz) [inline src/delay.cpp:91]: I would rather convert to `enum` and then use an `n_ports` enum (like in sledgehammer.cpp) in order to elliminate a constant, but I can also understand the argument of wanting a minimal change.

**Review by JohannesLorenz — APPROVED** [review]
Approving, though I would suggest to extend the commit message for the first commit from "Remove MSVC dllexport attribute" to something that also explains the background (why this has been done).

File: src/delay.cpp (hunk 1 above)
Comment (DomClark) [reply src/delay.cpp:91]: I agree that the `enum` approach is better, but I'm inclined to go for the minimal fix here. There seems to be a preference for the plugins using an enum to be in their own namespace, and the macro constants would need to be replaced with enumerators too. This is more than I was planning to do here, which is just an incidental change to fix this bug while I'm focusing on the MSVC support.

**Pull request merged**